From 6259f07c4a598edd11d3d25af10b3752e01b64b8 Mon Sep 17 00:00:00 2001
From: Dinesh Sudham <dineshx.sudham@intel.com>
Date: Fri, 24 Jul 2020 14:54:27 +0800
Subject: [PATCH] DRVLIB_SW-3754:XPCS: PHTSIDE SGMII configuration

GSW120 configured as MACSIDE SGMII,So XPCS should be
configure as PHY-SIDE SGMII, MAC_AUTO_SW set to zero
XPCS will operate at the speed/duplex after the
completion of CL37 ANEG

Signed-off-by: Dinesh Sudham <dineshx.sudham@intel.com>
---
 drivers/net/ethernet/lantiq/xpcs/xpcs.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/xpcs/xpcs.c b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
index e61071b6fcf2..d26e32724294 100644
--- a/drivers/net/ethernet/lantiq/xpcs/xpcs.c
+++ b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
@@ -1011,20 +1011,22 @@ static void xpcs_cl37_an(struct xpcs_prv_data *pdata)
 		/* 1 Full Duplex, 0 Half Duplex */
 		XPCS_RGWR_VAL(pdata, SR_MII_AN_ADV, FD, 1);
 		XPCS_RGWR_VAL(pdata, SR_MII_AN_ADV, HD, 0);
+		XPCS_RGWR_VAL(pdata, VR_MII_DIG_CTRL1, MAC_AUTO_SW, 0);
 	} else {
 		/* MAC Side SGMII */
 		XPCS_RGWR_VAL(pdata, VR_MII_AN_CTRL, TX_CONFIG, 0);
+		/* MAC side SGMII Automatic Speed Mode Change after CL37 AN,
+		 * xpcs automatically  switches to the negotiated
+		 * SGMII/USXGMII/QSGMII(port0) speed, SGMII get ANEG Config
+		 * from PHY about the link state change after the completion
+		 * of Clause 37 auto-negotiation between PHY and Link Partner
+		 */
+		 XPCS_RGWR_VAL(pdata, VR_MII_DIG_CTRL1, MAC_AUTO_SW, 1);
 	}
 
 	/* 4-BIT MII Interface */
 	XPCS_RGWR_VAL(pdata, VR_MII_AN_CTRL, MII_CONTROL, 0);
 
-	/* Automatic Speed Mode Change after CL37 AN, xpcs automatically
-	 * switches to the negotiated SGMII/USXGMII/QSGMII(port0) speed, after
-	 * the completion of Clause 37 auto-negotiation.
-	 */
-	XPCS_RGWR_VAL(pdata, VR_MII_DIG_CTRL1, MAC_AUTO_SW, 1);
-
 	/* 2.5G Mode */
 	if (pdata->mode == TWOP5G_SGMII_MODE) {
 		/* Link timer runs for 1.6 ms */
